rewrite-java/src/main/java/org/openrewrite/java/service/AnnotationService.java

```diff
@@ -53,7 +52,55 @@ public List<J.Annotation> getAllAnnotations(Cursor cursor) {
             return ((J.TypeParameters) j).getAnnotations();
         } else if (j instanceof J.Package) {
             return ((J.Package) j).getAnnotations();
+        } else if (j instanceof J.AnnotatedType) {
+            return getAllAnnotations((J.AnnotatedType) j);
+        } else if (j instanceof J.ArrayType) {
+            return getAllAnnotations((J.ArrayType) j);
+        } else if (j instanceof J.FieldAccess) {
+            return getAllAnnotations((J.FieldAccess) j);
+        } else if (j instanceof J.Identifier) {
+            return getAllAnnotations((J.Identifier) j);
+        }
+        return emptyList();
+    }
+
+    private List<J.Annotation> getAllAnnotations(J j) {
+        if (j instanceof J.AnnotatedType) {
+            return getAllAnnotations((J.AnnotatedType) j);
+        } else if (j instanceof J.ArrayType) {
+            return getAllAnnotations((J.ArrayType) j);
+        } else if (j instanceof J.Identifier) {
+            return getAllAnnotations((J.Identifier) j);
+        } else if (j instanceof J.FieldAccess) {
+            return getAllAnnotations((J.FieldAccess) j);
         }
         return emptyList();
     }
+
+    private List<J.Annotation> getAllAnnotations(J.AnnotatedType annotatedType) {
+        List<J.Annotation> annotations = new ArrayList<>(annotatedType.getAnnotations().size());
+        annotations.addAll(annotatedType.getAnnotations());
+        annotations.addAll(getAllAnnotations(annotatedType.getTypeExpression()));
+        return annotations;
+    }
+
+    private List<J.Annotation> getAllAnnotations(J.ArrayType arrayType) {
+        List<J.Annotation> annotations = new ArrayList<>(arrayType.getAnnotations() == null ? 0 : arrayType.getAnnotations().size());
+        annotations.addAll(getAllAnnotations(arrayType.getElementType()));
```

Comment:
This line should be removed. We are only interested in the annotations on the array type, not in any annotations on another type (including the array's element type). The idea behind the `AnnotationService` is just to provide an abstraction so that recipes don't have to know where to get the annotations from, as they can occur in multiple places, even when they pertain to the same element.

Reply:
Ah, I see -- so, it collects annotations that otherwise might be difficult to find like when the annotations are added in modifiers, class kind, a name, etc.

Reply:
Exactly. And in Kotlin it can be even trickier with use-site targets and multi annotations. At the same time this service allows recipes to be language independent to some extent (edits are at least not yet supported).